azuread/resource_application.go

```diff
@@ -624,6 +639,11 @@ func adApplicationSetOwnersTo(client graphrbac.ApplicationsClient, ctx context.C
 	ownersForRemoval := slices.Difference(existingOwners, desiredOwners)
 	ownersToAdd := slices.Difference(desiredOwners, existingOwners)
 
+	// add owners first to prevent a possible situation where terraform revokes its own access before adding it back.
```

Comment:
Would it be possible to get a test of this change? Or is it manual?

Reply:
Possibly but it would require a clientconfig datasource which we don't have in this provider yet & i don't think there is any need given switching the order logically fixes it